src/tally.cpp

```diff
@@ -420,59 +448,131 @@ std::string pyne::Tally::mcnp(int tally_index, std::string mcnp_version) {
+                                  axs, vec, origin, meshes, ints, e_bounds,
+                                  e_ints, out);
+
   } else {
-    std::cout << "tally/entity combination makes no sense for MCNP" << std::endl;
+    std::cout << "tally/entity combination makes no sense for MCNP"
+              << std::endl;
   }
-  
-
   // print sd card if area/volume specified
   return output.str();
-} 
+}
+
+template <typename T>
+bool pyne::Tally::is_zero(T vect) {
+  int size = sizeof(vect) / sizeof(vect[0]);
+  bool result = true;
+  for (int i = 0; i < size; i++) result &= (vect[i] == 0);
+  return result;
+}
+
 
 // Form the tally line as function of its properties
-std::string pyne::Tally::form_mcnp_tally(int tally_index, 
-                                               int type, 
-                                               std::string particle_token, 
-                                               int entity_id, double entity_size, 
-                                               double normalization) {
+std::string pyne::Tally::form_mcnp_tally(int tally_index, int type,
+                                         std::string particle_token,
+                                         int entity_id, double entity_size,
+                                         double normalization) {
   std::stringstream tally_stream;  // tally stream
   tally_stream << std::setiosflags(std::ios::fixed) << std::setprecision(6);
-  if (normalization != 1.0)
-    tally_stream << std::scientific;
-  
-  tally_stream << "F" << tally_index << type 
-               << ":" << particle_token << " " << entity_id << std::endl;
-  
+  if (normalization != 1.0) tally_stream << std::scientific;
+
+  tally_stream << "F" << tally_index << type << ":" << particle_token << " "
+               << entity_id << std::endl;
+
   if (entity_size > 0.0)
-    tally_stream << "SD" << tally_index << type << " " << entity_size << std::endl;
-  
+    tally_stream << "SD" << tally_index << type << " " << entity_size
+                 << std::endl;
+
   if (normalization != 1.0)
-    tally_stream << "FM" << tally_index << type << " " << normalization << std::endl;
+    tally_stream << "FM" << tally_index << type << " " << normalization
+                 << std::endl;
 
   return tally_stream.str();
 }
 
 
+// Form the mesh tally line as function of its properties
+std::string pyne::Tally::form_mcnp_meshtally(
+    int tally_index, std::string particle_token, std::string entity_geometry,
+    std::vector<double> axs, std::vector<double> vec,
+    std::vector<double> origin, std::vector<double> meshes[3],
+    std::vector<int> ints[3], std::vector<double> e_bounds,
+    std::vector<int> e_ints, std::string out) {
+  std::stringstream mtally_stream;
+  // indentation block
+  std::string indent_block = "           ";
+
+  mtally_stream << "FMESH" << tally_index << "4:" << particle_token << " ";
+  mtally_stream << "GEOM=";
+  std::stringstream sup_var;
```

Comment:
Why this variable? Why not append directly to `mtally_stream`?

Reply:
I wanted the tally card to be form in the same order:
GEOM, ORIGIN, MESH, INTS
(it is how it is explained in the MCNP docs, I am not sure if the order matters.)
if the order does not matter, I can append it directly to the `tally_stream`

Reply:
Order doesn't matter